clang/runtime/dpct-rt/include/util.hpp

```diff
@@ -939,6 +941,80 @@ class args_selector<n_nondefault_params, n_default_params, R(Ts...)> {
   }
 };
 
+// Supported group type during migration.
+enum class group_type { work_group, sub_group, logical_group, root_group };
```

Comment:
Let's put those new  logic into experimental:: namespace.  (1.logical_group is in experimental:: 2. https://github.com/intel/llvm/blob/c708b0d7321bee0e40f8a0a21173e705d7a5e2fc/sycl/doc/extensions/proposed/sycl_ext_oneapi_non_uniform_groups.asciidoc#L133 may be used instead of logical group.